packages/server/src/fhir/operations/execute.ts

```diff
@@ -167,10 +168,10 @@ export async function executeBot(request: BotExecutionRequest): Promise<BotExecu
       result = { success: false, logResult: 'Unsupported bot runtime' };
     }
   }
+  const executionTime = Number(process.hrtime.bigint() - execStart) / 1e9; // Report duration in seconds
 
-  const attributes = { project: bot.meta?.project, bot: bot.id };
-  incrementCounter('medplum.bot.execute', attributes);
-  incrementCounter(result.success ? 'medplum.bot.execute.success' : 'medplum.bot.execute.failure', attributes);
+  const attributes = { project: bot.meta?.project, bot: bot.id, outcome: result.success ? 'success' : 'failure' };
+  recordHistogramValue('medplum.bot.execution.time', executionTime, attributes);
```

Comment:
Consider `execute` over `execution` - Slight preference for aligning on using the FHIR Operation name for consistency purposes.

For example, considering these operations + metric names:

- `Bot/$deploy` -> `medplum.bot.deploy.time`
- `ValueSet/$expand` -> `medplum.valueset.expand.time`
- `Project/$expunge` -> `medplum.project.expunge.time`

Noun-izing might muddy those relationships.

Weakly held opinion.